cpp/include/mlt/metadata/stream.hpp

```diff
@@ -0,0 +1,246 @@
+#pragma once
+
+#include <mlt/common.hpp>
+#include <mlt/util/buffer_stream.hpp>
+#include <mlt/util/noncopyable.hpp>
+#include <mlt/util/varint.hpp>
+
+#include <optional>
+#include <memory>
+
+namespace mlt::metadata::stream {
+
+enum class DictionaryType {
```

Comment:
Maybe you want to use something like magic_enum? https://github.com/Neargye/magic_enum

Reply:
Why? We don't have need for the string equivalents so far, so I'm not seeing what value that would add to justify brining in more stuff.